charts/localstack/templates/deployment.yaml

```
@@ -120,8 +124,12 @@ spec:
             {{- end }}
             {{- if .Values.persistence.enabled }}
             - name: DATA_DIR
+            {{- if or (eq .Values.image.tag "latest") (not (hasPrefix .Values.image.tag "0.")) }}
+              value: /var/lib/localstack/data
+            {{- else }}
               value: /tmp/localstack/data
             {{- end }}
```

As above, here the order of the arguments for `hasPrefix` has to be reverted.
Also , as you can read in our docs [here](https://docs.localstack.cloud/references/filesystem/) and [here](https://docs.localstack.cloud/references/persistence-mechanism/), the `DATA_DIR` folder has been deprecated with `1.0`. Starting with `1.0`, the `PERSISTENCE` flag should be used and the folder will be `/var/lib/localstack/state`.

```suggestion
            {{- if hasPrefix "0." .Values.image.tag }}
            - name: DATA_DIR
              value: /tmp/localstack/data
            {{- else }}
            - name: PERSISTENCE
              value: "1"
            {{- end }}
```